docs/tutorial.rst

```diff
@@ -0,0 +1,172 @@
+.. role:: math(raw)
+   :format: html latex
+..
+
+.. role:: raw-latex(raw)
+   :format: latex
+..
+
+FPYLLL
+======
+
+::
+
+  >>> from fpylll import IntegerMatrix
+  >>> A = IntegerMatrix(9,10)
+  >>> A.randomize("intrel", bits=10)
+
+Matrix :math:`A` is a (random) knapsack type matrix. That is of the form  `[ {\bf a} | I_n]`, where `{\bf a}` is a column vector of dimension `n`, and `I_n` the `n`-dimensional identity matrix. Giving
+
+::
+
+  >>> print A
+  [ 116 1 0 0 0 0 0 0 0 0 ]
+  [ 331 0 1 0 0 0 0 0 0 0 ]
+  [ 303 0 0 1 0 0 0 0 0 0 ]
+  [ 963 0 0 0 1 0 0 0 0 0 ]
+  [ 456 0 0 0 0 1 0 0 0 0 ]
+  [ 225 0 0 0 0 0 1 0 0 0 ]
+  [ 827 0 0 0 0 0 0 1 0 0 ]
+  [ 381 0 0 0 0 0 0 0 1 0 ]
+  [  99 0 0 0 0 0 0 0 0 1 ]
+
+    
+Also, the following type of matrices are supported,
+
+::
+
+  >>> b = 10
+  >>> p = 521 # prime
+  >>> A = IntegerMatrix(10,10)
+  >>> A.randomize("uniform", bits=b)
+  >>> A.randomize("ntrulike", bits=b, q=p)
+  >>> A.randomize("ntrulike2", bits=b, q=p)
+
+For a user defined matrix we use the method **from\_matrix**
+
+::
+
+  >>> A = IntegerMatrix.from_matrix([[1,2,3,4],[30,4,4,5],[1,-2,3,4],[0,0,1,0]])
+
+
+  To compute the Gram-Schimdt form of the matrix `{\bf A}`, we use the **GSO** class
+
+::
+
+  >>> from fpylll import GSO
+  >>> M = GSO.Mat(A)
+  >>> # to write a vector v as a linear combination of the GS-basis of RowSp(A):
```

Comment:
I'd write this as normal text instead of a comment in a code block, i.e. break up the code block in two